src/command.cpp

```diff
@@ -360,6 +362,85 @@ static const Command _command_proc_table[] = {
 	DEF_CMD(CmdOpenCloseAirport,                               0, CMDT_ROUTE_MANAGEMENT      ), // CMD_OPEN_CLOSE_AIRPORT
 };
 
+
+/**
+ * List of flags for a command log entry
+ */
+enum CommandLogEntryFlagEnum {
+	CLEF_NONE                = 0x00, ///< no flag is set
+	CLEF_CMD_FAILED          = 0x01, ///< command failed
+	CLEF_GENERATING_WORLD    = 0x02, ///< generating world
+	CLEF_TEXT                = 0x04, ///< have command text
+	CLEF_ESTIMATE_ONLY       = 0x08, ///< estimate only
+	CLEF_ONLY_SENDING        = 0x10, ///< only sending
+	CLEF_MY_CMD              = 0x20, ///< locally generated command
+};
+DECLARE_ENUM_AS_BIT_SET(CommandLogEntryFlagEnum)
+typedef SimpleTinyEnumT<CommandLogEntryFlagEnum, byte> CommandLogEntryFlag;
```

Comment:
I think C++11 enum base type declarations are acceptable to use now, i.e. declaring `enum: CommandLogEntryFlag : byte { CLEF_NONE ... };` so separate enum and storage types are not required.

Reply:
I also wonder if we should start using enum classes.

Reply:
Definitely change this to enum with underlying type ahead of #7538